src/status_im/ui/screens/activity_center/views.cljs

```diff
@@ -107,24 +111,64 @@
 (defn notifications-list
   []
   (let [notifications (<sub [:activity-center/notifications-per-read-status])]
-    [rn/flat-list {:style          {:padding-horizontal 8}
-                   :data           notifications
+    [rn/flat-list {:data           notifications
                    :key-fn         :id
                    :on-end-reached #(>evt [:activity-center.notifications/fetch-next-page])
                    :render-fn      render-notification}]))
 
+(defn filter-selector-read []
+  (let [unread-filter-enabled?  (<sub [:activity-center/status-filter-unread-enabled?])]
+    ;; TODO: Replace the button by a Filter Selector component once available for use.
+    [button/button {:icon     true
+                    :type     (if unread-filter-enabled? :primary :outline)
+                    :size     32
+                    :on-press #(if unread-filter-enabled?
+                                 (>evt [:activity-center.notifications/fetch-first-page {:status-filter :read}])
```

Comment:
no need to duplicate events , `#(>evt [:activity-center.notifications/fetch-first-page {:status-filter (if unread-filter-enabled? :read :unread}])` reads well enough

Reply:
or you could move status-filter to let

Reply:
Good suggestion. I'm marking this thread as resolved because I implemented this in another branch I'm currently working on.